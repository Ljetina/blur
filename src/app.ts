import 'dotenv/config';
import express, { Express } from 'express';
import http from 'http';

import { addAuthRoutes, authenticate, getSessionMiddleWare } from './lib/auth';
import { initialDataHandler } from './routes/initial/initial';
import {
  handleCreateConversation,
  handleDeleteConversation,
  handleUpdateConversation,
} from './routes/conversation/conversation';
import { handleUpdateUserPreferences } from './routes/user/user';
import { morganMiddleware } from './lib/log';
import { handleWaitingList } from './routes/waitinglist/waitinglist';
import { handleMessages } from './routes/conversation/messages/messages';
import { handleDemoConversation, handleDemoPricing } from './routes/demo/demo';
import {
  handleGetConversationNotebookSettings,
  handleUpsertConversationNotebook,
} from './routes/conversation/notebook/notebook';
import { handleCreateUserNotebook } from './routes/user/notebook/notebook';
import { startWsServer } from './stream/wschat';

export function prepareApp(sessionMiddleware: any) {
  const app = express();

  app.use(express.json());
  app.use(morganMiddleware);
  addAuthRoutes(app, sessionMiddleware);
  app.get('/initial', authenticate, initialDataHandler);
  app.post('/conversation', authenticate, handleCreateConversation);
  app.delete('/conversation/:id', authenticate, handleDeleteConversation);
  app.put('/conversation/:id', authenticate, handleUpdateConversation);
  app.get('/conversation/:id/message', authenticate, handleMessages);
  app.post(
    '/conversation/:id/notebook',
    authenticate,
    handleUpsertConversationNotebook
  );
  app.get(
    '/conversation/:id/notebook',
    authenticate,
    handleGetConversationNotebookSettings
  );
  app.put('/user', authenticate, handleUpdateUserPreferences);
  app.post('/user/notebook', authenticate, handleCreateUserNotebook);
  app.post('/waitinglist', handleWaitingList);

  app.get('/demo/main', handleDemoConversation);
  app.get('/demo/pricing', handleDemoPricing);

  return app;
}

export function startListening() {
  const app = prepareApp(getSessionMiddleWare());
  const server = http.createServer(app);
  startWsServer(server);
  server.listen(
    process.env.NODE_ENV == 'test'
      ? 9999
      : process.env.PORT
      ? JSON.parse(process.env.PORT)
      : 3001
  );
}
