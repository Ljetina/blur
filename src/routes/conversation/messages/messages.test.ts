import request from 'supertest';
import { prepareApp } from '../../../app.js'; // Import your Express app
import { describe, it, expect } from 'vitest';

describe('GET /conversations/:id/messages', () => {
  it('should return empty response', async () => {
    const res = await request(prepareApp())
      .get(`/conversation/a319526b-4732-45dc-b8d0-a3fc5836179c/message`)
      .set('x-inject-user_id', '2b802813-86f0-4130-a06b-7e1775350592')
      .set('x-inject-tenant_id', 'ec224ff1-0f67-4164-8d06-37692f134c3a');

    expect(res.text).toMatchSnapshot();
  });
});

export {};
