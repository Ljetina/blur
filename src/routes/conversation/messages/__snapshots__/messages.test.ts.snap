// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GET /conversations/:id/messages > should return empty response 1`] = `"{\\"data\\":[],\\"pagination\\":{\\"total_pages\\":0,\\"total_records\\":0}}"`;
